Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/IPay88TokenizationViewModel.swift

```diff
@@ -0,0 +1,547 @@
+            }
+        }
+        
+        // Customer validation
+        
+        if PrimerAPIConfiguration.current?.clientSession?.customer?.firstName == nil {
+            let err = PrimerError.invalidClientSessionValue(
+                name: "customer.firstName",
+                value: nil,
+                userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"],
+                diagnosticsId: nil)
+            ErrorHandler.handle(error: err)
+            errors.append(err)
+        }
+        
+        if PrimerAPIConfiguration.current?.clientSession?.customer?.lastName == nil {
+            let err = PrimerError.invalidClientSessionValue(
+                name: "customer.lastName",
+                value: nil,
+                userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"],
+                diagnosticsId: nil)
+            ErrorHandler.handle(error: err)
+            errors.append(err)
+        }
+        
+        if PrimerAPIConfiguration.current?.clientSession?.customer?.emailAddress == nil {
+            let err = PrimerError.invalidClientSessionValue(
+                name: "customer.emailAddress",
+                value: nil,
+                userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"],
+                diagnosticsId: nil)
+            ErrorHandler.handle(error: err)
+            errors.append(err)
+        }
+        
+        if PrimerAPIConfiguration.current?.clientSession?.customer?.mobileNumber == nil {
+            let err = PrimerError.invalidClientSessionValue(
+                name: "customer.mobileNumber",
+                value: nil,
+                userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"],
+                diagnosticsId: nil)
+            ErrorHandler.handle(error: err)
+            errors.append(err)
+        }
+        
+#if !canImport(PrimerIPay88SDK)
+        let err = PrimerError.missingSDK(
+            sdkName: "PrimerIPay88SDK",
+            userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"],
+            diagnosticsId: nil)
+        ErrorHandler.handle(error: err)
+        errors.append(err)
+#endif
+        
+        if errors.count == 1 {
+            throw errors.first!
+            
+        } else if errors.count > 1 {
+            let err = PrimerError.underlyingErrors(
+                errors: errors,
+                userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"],
+                diagnosticsId: nil)
+            ErrorHandler.handle(error: err)
+            throw err
+        }
+    }
+    
+    override func performPreTokenizationSteps() -> Promise<Void> {
+        let event = Analytics.Event(
+            eventType: .ui,
+            properties: UIEventProperties(
+                action: .click,
```

Comment:
Don't think that's the right place to call this.
We would be logging a `.click` event in headless checkout, which doesn't make much sense here 🤔

Reply:
Yeah, you're right